graphbolt/src/cnumpy.h

```diff
@@ -0,0 +1,71 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file cnumpy.h
+ * @brief Numpy File Fetecher class.
+ */
+
+#include <fcntl.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <torch/script.h>
+
+#ifdef __linux__
+#include <liburing.h>
+#include <unistd.h>
+#endif
+
+#include <cassert>
+#include <cstdio>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+namespace graphbolt {
+namespace storage {
+
+/**
+ * @brief Disk Numpy Fetecher class.
+ */
+class OnDiskNpyArray : public torch::CustomClassHolder {
+ public:
+  /** @brief Default constructor. */
+  OnDiskNpyArray() = default;
+
+  /** @brief Constructor with given file path. */
+  OnDiskNpyArray(std::string filename, torch::ScalarType dtype);
+
+  /** @brief Deconstructor. */
+  ~OnDiskNpyArray();
+
+  /**
+   * @brief Parse numpy meta data.
+   */
+  void ParseNumpyHeader();
+
+  /**
+   * @brief Read disk numpy file based on given index and transform to
+   * tensor.
+   */
+  torch::Tensor IndexSelectIOuring(torch::Tensor idx);
+
+ private:
+  std::string filename_;           // Path to numpy file.
+  torch::ScalarType dtype_;        // Feature data type.
+  int feature_fd_;                 // File description.
```

Comment:
I would rename this to `file_description_` since it does not really related to feature, just a file property.